### ISSUE TITLE
supporting google maps

### ISSUE THREAD

**Issue by zcaudate** [sothawo/mapjfx#49]
I noticed that there is a open layers google maps library here: https://github.com/mapgears/ol3-google-maps

@sothawo, I'd like to get my hands dirty but I'm just starting to test out the mapjfx apis. By your estimate, would integrating google maps be hard to do?

**Comment by sothawo** [comment]
this sure looks interesting; I'm not sure if the existing handling of clickes and events will work without major changes; this is something to be tried and tested.
What's harder is the fact, that the Google Mpas Api Key must be present in the HTML before the normal Javascript will be loaded, that means that in a Java application, after entering the Google Maps API Key it would be necessary to completely build a new MapView object.
I'll mark this ans enhancement, but I don't know when I might find time for this.

**Comment by zcaudate** [comment]
Are you able to point to the locations of code that you are talking about. How different is supporting Google Maps as opposed to Bing Maps?

**Comment by sothawo** [comment]
Bing Maps integration is provided by OL, so my code only uses OL functions, whereas the GM integration - as far as I understood it at a first glance - sits on top of the OL code.
So it might not be too hard to change the code to use GM, but as a library mapjfx needs to support all the different map types and - in the best case - switching between them without loosing markers, labels etc.
I have to take a deeper look at this to find the exact places to change/adapt.

**Comment by zcaudate** [comment]
okay. thanks for the explanation.

**Comment by sothawo** [comment]
I'll have a look at this now; let's see if it can be integrated

**Comment by sothawo** [comment]
I tried to integrate olgm (created a new branch _google-maps-support_ branched of 2.x), but found the following problems:

- olgm comes bundled with a version of OpenLayers, so I have to remove the version I use. The latest version of olgm 1.1.1 from 2019/02/20 includes ol version 5.0.3 from  2018/07/03. The current version of ol, that I implemented mapjfx against, is 5.3.0 from 2018/11/06, so my ol calls which should be handled by olgm, are not compatible, I guess this is the main cause for the other problems.
- I managed to integrate olgm into the test html I use and the lib and was able to create a view with Google Maps as layer. But this had several problems:
- The map does not fill the entire space, there is a grey area at the bottom, see attached screenshot
- There are no zoom controls visible
- When showing a marker and then using the scroll wheel to zoom, the marker jumps to a different position although the mouse cursor is not even near to it
- When including olgm and then using OSM map style, the map view, marker etc are working as expected, but no zoom controls or attributions are shown.

This might be worth a second try when olgm uses an up to date version of ol, I'll keep the branch around on the repo.

![bildschirmfoto 2019-03-02 um 18 00 47](https://user-images.githubusercontent.com/8456476/53685195-c2028d00-3d17-11e9-9443-85a10eb4899c.png)
